core/src/components/cat-modal/cat-modal.tsx

```diff
@@ -0,0 +1,78 @@
+import { Component, h, Method, Prop } from '@stencil/core';
+import * as focusTrap from 'focus-trap';
+import firstTabbable from '../../utils/first-tabbable';
+
+@Component({
+  tag: 'cat-modal',
+  styleUrl: 'cat-modal.scss',
+  shadow: true
+})
+export class CatModal {
+  private trap?: focusTrap.FocusTrap;
+  private modal?: HTMLElement;
+  private modalWrapper?: HTMLElement;
+  private closeButton?: HTMLElement;
+
+  /**
+   * The size of the modal.
+   */
+  @Prop() size: 's' | 'm' | 'l' = 'm';
+
+  componentDidLoad() {
+    if (this.modal) {
+      this.trap = focusTrap.createFocusTrap(this.modal, {
+        tabbableOptions: {
+          getShadowRoot: true
+        },
+        initialFocus: firstTabbable(this.closeButton),
+        allowOutsideClick: true,
+        clickOutsideDeactivates: event => !this.modal || !event.composedPath().includes(this.modal),
+        onDeactivate: () => this.modalWrapper?.classList.remove('is-visible'),
+        setReturnFocus: previousActiveElement =>
+          previousActiveElement instanceof HTMLElement
+            ? (firstTabbable(previousActiveElement) as HTMLElement)
+            : previousActiveElement
+      });
+    }
+  }
+
+  /**
+   * Shows the modal.
+   */
+  @Method()
+  async show() {
+    this.modalWrapper?.classList.add('is-visible');
```

Comment:
I think a better way is to have a state of the `show` and use that state in a `class` attribute
for example: `class={{show: this.show}}`
This is a much cleaner way IMHO

Reply:
Ok, I'll take a look at it, it also seems to me that it would be cleaner, thanks for commenting 😊